dist-tracing/src/trace.rs

```diff
@@ -0,0 +1,255 @@
+use chrono::{DateTime, Utc};
+use tracing_subscriber::registry::LookupSpan;
+
+// TODO: review pub vs. pub(crate)
+#[derive(PartialEq, Eq, Hash, Clone, Debug)]
+pub struct TraceCtx {
+    pub parent_span: Option<SpanId>,
+    pub trace_id: TraceId,
+}
+
+// todo extend error and etc
+#[derive(PartialEq, Eq, Hash, Clone, Debug)]
+pub enum TraceCtxError {
+    TelemetryLayerNotRegistered,
+    RegistrySubscriberNotRegistered,
+    NoEnabledSpan,
+    NoParentNodeHasTraceCtx, // no parent node has explicitly registered trace ctx
+}
```

Comment:
is this public? probably don't want to export a public enum, since exhaustive enum matching is a forward compatibility hazard (adding new variants is a breaking change). I would either make the error enum private and hide it behind a struct, or put the `#[non_exhaustive]` attribute on this.